docs/reference/pc_messages.md

```diff
@@ -0,0 +1,43 @@
+---
+title: "Package Control: Messages"
+---
+
+<!-- https://packagecontrol.io/docs/messaging -->
+<!-- https://github.com/wbond/packagecontrol.io/blob/master/app/html/docs/messaging.html -->
+<!-- https://github.com/wbond/package_control/blob/master/example-messages.json -->
+
+
+# Messages
+
+It is possible to show users a (plain text) message
+when the package is installed,
+and after upgrades to specific versions of the package.
+
+This is controlled by a `messages.json` file in the root of the package.
+See below for an example of the proper structure of the JSON.
+Each value will be a file path that is relative to the package root.
+Each key will either be the string "install" or a version number.
+
+When a package is **installed**,
+if the key "install" is present in the messages.json file,
+the file at the specified path will be displayed to the user.
+
+When a package is **upgraded**,
+Package Control looks through each key in the messages.json file
+and shows the content of the text file that is a value of any key
+that is higher than the previous version of the package the user had installed.
+Thus if the user had version 1.1.0 installed,
+the files for 1.2.0 and 1.1.1 would be shown.
+If the user had version 1.1.1 installed,
+only the messages for version 1.2.0 would be shown.
+
+
+## Example
+
+```json
+{
+	"install": "messages/install.txt",
+	"1.1.1": "messages/1.1.1.txt",
+	"1.2.0": "messages/1.2.0.txt"
+}
```

Comment:
```suggestion
```json
{
	"1.2.0": "messages/1.2.0.txt",
	"1.1.1": "messages/1.1.1.txt",
	"install": "messages/install.txt"
}
```
Reverse-order is convenient for (a) Git diffs when appending, and (b) that's the way Changelog.md suggests.

Reply:
What's the Changelog.md you're referring to?

Reply:
https://keepachangelog.com, where you have an "Unreleased" section at the top that is renamed to a version and then a new "Unreleased" stacked on top. It pushes down the older versions, resulting in reverse-release order. See also: GitHub releases and OpenSSH changelog.

No, I've never had trouble diffing a messages.json. But I am annoyed by the no-trailing-comma rule requiring an additional line change every time you add a final item to arrays or objects.

Reply:
Right, if you view this as a changelog newest on top makes sense. I can go either way here, we can recommend that, it's fine. Not something I'd personally ever do or would actively push for in a PR review though.